src/software/primitive/primitive_msg_factory.cpp

```diff
@@ -0,0 +1,160 @@
+#include "software/primitive/primitive_msg_factory.h"
+
+std::unique_ptr<ChipPrimitiveMsg> createChipPrimitiveMsg(const Point &chip_origin,
+                                                         const Angle &chip_direction,
+                                                         float chip_distance_meters)
+{
+    auto chip_primitive_msg = std::make_unique<ChipPrimitiveMsg>();
+
+    auto chip_origin_msg = std::make_unique<PointMsg>();
+    chip_origin_msg->set_x_meters(static_cast<float>(chip_origin.x()));
+    chip_origin_msg->set_y_meters(static_cast<float>(chip_origin.y()));
+    *(chip_primitive_msg->mutable_chip_origin()) = *chip_origin_msg;
+
+    auto chip_direction_msg = std::make_unique<AngleMsg>();
+    chip_direction_msg->set_radians(static_cast<float>(chip_direction.toRadians()));
+    *(chip_primitive_msg->mutable_chip_direction()) = *chip_direction_msg;
```

Comment:
I think you can make use of some of the translation functions available in `software/proto/message_translation`
Specifically these ones found in `tbots_protobuf.h`
![image](https://user-images.githubusercontent.com/31548846/88871956-ff4da180-d1cd-11ea-9670-253590805787.png)

Ditto for all the other functions

Reply:
Will save me some time in the future. Now using it.